src/udp_wrap.cc

```diff
@@ -446,6 +446,10 @@ void UDPWrap::OnRecv(uv_udp_t* handle,
   }
 
   char* base = static_cast<char*>(realloc(buf->base, nread));
+  // If realloc fails, fallback to old pointer
+  if(base == NULL && nread != 0)
+      base = buf->base;
```

Comment:
If realloc() failed then `buf->base its original size, which is *not* `nread`, so this "fix" means passing a buffer to `Buffer::Use()` that is smaller than it is claimed to be, which is a memory error.

What problem are you trying to solve?

Note that memory allocation failures should be fatal, see line 310 above. With modern O/S and their memory overcommitment, its pretty hard to get allocs of small memory areas to fail, btw.